src/einsteinpy/hypersurface/embedding.py

```diff
@@ -0,0 +1,188 @@
+      """
+      Returns spherical radial coordinate (of the embedding) from given schwarzschild
+      coordinate.
+
+      Parameters
+      ----------
+      M : ~astropy.units.kg
+          Mass of the body
+      r : float
+
+      Returns
+      -------
+      float : spherical radial coordinate of the 3d embedding
+
+      """
+      return r / math.sqrt(1 - (2 * M.value / r))
+
+
+   def get_values(self, M, alpha):
+      """
+      Obtain the Z coordinate values and corrosponding R values for range of
+      r as 9m/4 < r < 9m.
+
+      Parameters
+      ----------
+      M : ~astropy.units.kg
+          Mass of the body
+      alpha : float
+              scaling factor to obtain the step size for incrementing r
+
+      Returns
+      -------
+      tuple
+          (list, list) : values of R (x_axis) and Z (y_axis)
+
+      """
+      x_axis = []
+      y_axis = []
+      r_initial = self.r_init.value
+      r_step = M.value / alpha
+
+      z = 0
+      r = r_initial
+      while (r < 9 * M.value):
+          x_axis.append(self.radial_coord(M, r))
+          y_axis.append(z)
+          z = z + self.gradient(M, r) * r_step
+          r = r + r_step
+
+      z = 0
+      r = r_initial
+      while (r > (9 * M.value / 4)):
+          x_axis.append(self.radial_coord(M, r))
+          y_axis.append(z)
+          z = z + self.gradient(M, r) * r_step
+          r = r - r_step
+
+      return x_axis, y_axis
+
+   def get_values_surface(self, M, alpha):
+      """
+      Obtain the same values as of the above function but reshapes them to obtain
+      values for all points on the solid of revolution about Z axis (as the
+      embedding is symmetric in angular coordinates).
+
+      Paramters
+      ---------
+      M : ~astropy.units.kg
+      alpha : float
+              scaling factor to obtain the step size for incrementing r
+
+      Returns
+      -------
+      tuple
+          (~numpy.array of X, ~numpy.array of Y, ~numpy.array of Z) : values in cartesian coordinates
+          obtained after applying solid of revolution
+
+      """
+      r_initial = self.r_init.value
+      r_step = M.value / alpha
+      phi_values = np.linspace(0, 2*np.pi, 60)
+      R_values = []
+      z_values = []
+
+      z = 0
+      r = r_initial
+      while (r < 20 * M.value):
+          R_values.append(self.radial_coord(M, r))
+          z_values.append(z)
+          z = z + self.gradient(M, r) * r_step
+          r = r + r_step
+
+      R_values = np.array(R_values)
+      R_values, phi_values = np.meshgrid(R_values, phi_values)
+
+      X = R_values * np.cos(phi_values)
+      Y = R_values * np.sin(phi_values)
+      x_len = X.shape[0]
+      Z = np.array(z_values)
+      z_values = np.array(z_values)
+      for i in range(0, x_len - 1):
+          Z = np.concatenate((Z, z_values), axis = 0)
+
+      Z.reshape((X.shape[0], X.shape[1]))
+      return X, Y, Z
+
+
+   def plot_hypersurface(self, type='wireframe', alpha=100):
```

Comment:
This can go to plotting. Put this function in a GitHub Gist and give the llink to me.